packages/ipfs-core/src/components/pin/ls.js

```diff
@@ -19,13 +19,47 @@ function toPin (type, cid, metadata) {
   return output
 }
 
+/**
+ * @param {Object} config
+ * @param {import('./pin-manager')} config.pinManager
+ * @param {import('../index').DAG} config.dag
+ */
 module.exports = ({ pinManager, dag }) => {
-  return withTimeoutOption(async function * ls (options = {}) {
+  /**
+   * List all the objects pinned to local storage
+   *
+   * @param {LsOptions} [options]
+   * @returns {AsyncIterable<LsEntry>}
+   * @example
+   * ```js
+   * for await (const { cid, type } of ipfs.pin.ls()) {
+   *   console.log({ cid, type })
+   * }
+   * // { cid: CID(Qmc5XkteJdb337s7VwFBAGtiaoj2QCEzyxtNRy3iMudc3E), type: 'recursive' }
+   * // { cid: CID(QmZbj5ruYneZb8FuR9wnLqJCpCXMQudhSdWhdhp5U1oPWJ), type: 'indirect' }
+   * // { cid: CID(QmSo73bmN47gBxMNqbdV6rZ4KJiqaArqJ1nu5TvFhqqj1R), type: 'indirect' }
+   *
+   * const paths = [
+   *   CID.from('Qmc5..'),
+   *   CID.from('QmZb..'),
+   *   CID.from('QmSo..')
+   * ]
+   * for await (const { cid, type } of ipfs.pin.ls({ paths })) {
+   *   console.log({ cid, type })
+   * }
+   * // { cid: CID(Qmc5XkteJdb337s7VwFBAGtiaoj2QCEzyxtNRy3iMudc3E), type: 'recursive' }
+   * // { cid: CID(QmZbj5ruYneZb8FuR9wnLqJCpCXMQudhSdWhdhp5U1oPWJ), type: 'indirect' }
+   * // { cid: CID(QmSo73bmN47gBxMNqbdV6rZ4KJiqaArqJ1nu5TvFhqqj1R), type: 'indirect' }
+   * ```
+   */
+  async function * ls (options = {}) {
+    /** @type {PinQueryType} */
     let type = PinTypes.all
 
     if (options.type) {
       type = options.type
       if (typeof options.type === 'string') {
+        // @ts-ignore - can't infer toLowerCase()
         type = options.type.toLowerCase()
```

Comment:
If you did something like:

```js
`${options.type}`.toLowerCase()
```

maybe you wouldn't need the @ts-ignore comment?

Reply:
@achingbrain Problem here is not that TS can't infer a string (it does because there is a typeof check) but the fact that `toLowerCase()` returns value that has a string subtype of `PinQueryType` which is enum of specific strings constants.

I'll refine my comment there to be more clear about it